src/test/java/com/connexta/search/common/SearchITests.java

```diff
@@ -7,430 +7,543 @@
+      return new URL("http", "localhost", 9983, "/solr");
+    }
+  }
 
-  private static final String INDEX_ENDPOINT_BASE_URL = "/index/";
+  @Inject
+  private SolrClient solrClient; // Injected to empty solr for each test. Not tested directly.
 
-  @NotNull
-  private static final Map<String, String> getSampleDatatHavingAllAttributes =
-      Map.of(
-          ID_ATTRIBUTE_NAME,
-          "00067360b70e4acfab561fe593ad3f7b",
-          CONTENTS_ATTRIBUTE_NAME,
-          "Winterfell",
-          MEDIA_TYPE_ATTRIBUTE_NAME,
-          "application/json");
-
-  private static final String allAttributesQuery =
-      QUERY_TERMS.stream()
-          .map(
-              term -> String.format("%s = '%s'", term, getSampleDatatHavingAllAttributes.get(term)))
-          .collect(Collectors.joining(" AND "));
-
-  @Inject private QueryManager queryManager;
-  @Inject private IndexManager indexManager;
-  @Inject private TestRestTemplate restTemplate;
-  @Inject private SolrClient solrClient;
-
-  @Value("${endpointUrl.productRetrieve}")
-  private String productRetrieveEndpoint;
+  @Inject private WebTestClient webTestClient;
 
   @Value("${endpoints.index.version}")
   private String indexApiVersion;
 
+  private static MockWebServer storeMockWebServer;
+
   @BeforeEach
   public void beforeEach() throws IOException, SolrServerException {
+    storeMockWebServer = new MockWebServer();
+    storeMockWebServer.start();
+
     // TODO shouldn't need to clear solr every time
     solrClient.deleteByQuery(SOLR_COLLECTION, "*");
     solrClient.commit(SOLR_COLLECTION);
   }
 
-  @Test
-  public void testContextLoads() {}
-
-  @ParameterizedTest
-  @ValueSource(strings = {"", " ", "text"})
-  public void testStoringValidCst(String contents) throws Exception {
-    // given
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-
-    // when indexing a product
-    ResponseEntity<String> response =
-        restTemplate.exchange(
-            indexEndpointUrl,
-            HttpMethod.PUT,
-            createIndexRequest("{ \"ext.extracted.text\" : \"" + contents + "\" }"),
-            String.class);
-
-    // then
-    assertThat(response.getStatusCode(), is(HttpStatus.OK));
+  @AfterEach
+  public void afterEach() throws IOException {
+    storeMockWebServer.shutdown();
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = {"", "{}", "{ \"\": \"text\"}"})
-  public void testStoringInvalidCst(String contents) throws IOException {
-    // given
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-    // when indexing a product
-    ResponseEntity<String> response =
-        restTemplate.exchange(
-            indexEndpointUrl, HttpMethod.PUT, createIndexRequest(contents), String.class);
-
-    // then
-    assertThat(response.getStatusCode(), is(HttpStatus.INTERNAL_SERVER_ERROR));
-  }
+  @Test
+  public void testContextLoads() {}
```

Comment:
✏️ This test is useless. Every test already tests this by just running.

Reply:
We have `testContextLoads` in all of our test classes. Should we go ahead and remove them everywhere across all 3 repos?